db/db_options_test.cc

```diff
@@ -266,6 +267,44 @@ TEST_F(DBOptionsTest, SetMutableTableOptions) {
   ASSERT_EQ(c_bbto->block_restart_interval, 13);
 }
 
+TEST_F(DBOptionsTest, SetWithCustomMemTableFactory) {
+  class DummySkipListFactory : public SkipListFactory {
+   public:
+    static const char* kClassName() { return "DummySkipListFactory"; }
+    const char* Name() const override { return kClassName(); }
+    explicit DummySkipListFactory(size_t lookahead = 0)
+        : SkipListFactory(lookahead) {}
+  };
+  Options options;
+  options.create_if_missing = true;
+  options.fail_if_options_file_error = true;
+  options.env = env_;
+  options.blob_file_size = 16384;
+  options.disable_auto_compactions = false;
+  ConfigOptions config_options;
+  config_options.ignore_unsupported_options = false;
```

Comment:
Not entirely following this. If we set `ignore_unsupported_options` to `false` and `fail_if_options_file_error` to `true`, how do the operations below like `SetOptions` or CF creation/drop succeed?

Reply:
The ignore unsupported options here is to validate that we are using a MemTableFactory not registered with the repo and does not affect the SetOptions et al calls.

The fail_if_options_file_error causes errors (such as validation) to be bubbled up in the returned status from operations such as SetOptions.  If this value is false, failure to write an options file will still be success.  If false, it will return success even on failure to write the file.  In this case, we should succeed under both circumstances (so false is more important).  The code has been restructured to test both values.